x11-packages/qt6-qtbase/build.sh

```diff
@@ -0,0 +1,133 @@
+TERMUX_PKG_BUILD_DEPENDS="libwayland-protocols, vulkan-headers, vulkan-loader-generic"
+TERMUX_PKG_HOSTBUILD=true
+TERMUX_PKG_FORCE_CMAKE=true
+TERMUX_PKG_NO_STATICSPLIT=true
+TERMUX_PKG_EXTRA_CONFIGURE_ARGS="
+-DCMAKE_INTERPROCEDURAL_OPTIMIZATION=ON
+-DCMAKE_MESSAGE_LOG_LEVEL=STATUS
+-DCMAKE_SYSTEM_NAME=Linux
+-DFEATURE_journald=OFF
+-DFEATURE_no_direct_extern_access=ON
+-DFEATURE_openssl_linked=ON
+-DFEATURE_system_sqlite=ON
+-DINSTALL_ARCHDATADIR=lib/qt6
+-DINSTALL_BINDIR=lib/qt6/bin
+-DINSTALL_DATADIR=share/qt6
+-DINSTALL_DOCDIR=share/doc/qt6
+-DINSTALL_EXAMPLESDIR=share/doc/qt6/examples
+-DINSTALL_INCLUDEDIR=include/qt6
+-DINSTALL_LIBEXECDIR=lib/qt6
+-DINSTALL_MKSPECSDIR=lib/qt6/mkspecs
+-DINSTALL_PUBLICBINDIR=${TERMUX_PREFIX}/bin
+-DQT_ALLOW_SYMLINK_IN_PATHS=OFF
+-DQT_FEATURE_freetype=ON
+-DQT_FEATURE_gui=ON
+-DQT_FEATURE_harfbuzz=ON
+-DQT_FEATURE_widgets=ON
+-DQT_FEATURE_zstd=ON
+-DQT_HOST_PATH=${TERMUX_PREFIX}/opt/qt6/cross
+"
+TERMUX_PKG_NO_SHEBANG_FIX_FILES="
+lib/qt6/bin/qmake
+lib/qt6/bin/qmake6
+lib/qt6/bin/qt-cmake
+lib/qt6/bin/qt-cmake-create
+lib/qt6/bin/qt-configure-module
+lib/qt6/bin/qtpaths
+lib/qt6/bin/qtpaths6
+opt/qt6/cross/lib/cmake/Qt6/libexec/qt-internal-ninja.in
+opt/qt6/cross/lib/cmake/Qt6/libexec/qt-internal-strip.in
+opt/qt6/cross/lib/qt6/bin/qt-cmake
+opt/qt6/cross/lib/qt6/bin/qt-cmake-create
+opt/qt6/cross/lib/qt6/bin/qt-configure-module
+opt/qt6/cross/lib/qt6/mkspecs/features/data/mac/objc_namespace.sh
+opt/qt6/cross/lib/qt6/mkspecs/features/uikit/device_destinations.sh
+opt/qt6/cross/lib/qt6/mkspecs/features/uikit/devices.py
+opt/qt6/cross/lib/qt6/qt-cmake-private
+opt/qt6/cross/lib/qt6/qt-cmake-standalone-test
+opt/qt6/cross/lib/qt6/qt-internal-configure-examples
+opt/qt6/cross/lib/qt6/qt-internal-configure-tests
+opt/qt6/cross/lib/qt6/qt-testrunner.py
+opt/qt6/cross/lib/qt6/sanitizer-testrunner.py
+"
+TERMUX_PKG_RM_AFTER_INSTALL="
+lib/objects-*
+opt/qt6/cross/lib/objects-*
+"
+
+termux_step_host_build() {
+	termux_setup_cmake
+	termux_setup_ninja
+
+	cmake \
+		-G Ninja \
+		-S ${TERMUX_PKG_SRCDIR} \
+		-DCMAKE_BUILD_TYPE=MinSizeRel \
+		-DCMAKE_INSTALL_PREFIX=${TERMUX_PREFIX}/opt/qt6/cross \
+		-DCMAKE_INTERPROCEDURAL_OPTIMIZATION=ON \
+		-DCMAKE_MESSAGE_LOG_LEVEL=STATUS \
+		-DFEATURE_journald=OFF \
+		-DFEATURE_openssl_linked=ON \
+		-DFEATURE_system_sqlite=ON \
+		-DINSTALL_ARCHDATADIR=lib/qt6 \
+		-DINSTALL_BINDIR=lib/qt6/bin \
+		-DINSTALL_DATADIR=share/qt6 \
+		-DINSTALL_DOCDIR=share/doc/qt6 \
+		-DINSTALL_EXAMPLESDIR=share/doc/qt6/examples \
+		-DINSTALL_INCLUDEDIR=include/qt6 \
+		-DINSTALL_LIBEXECDIR=lib/qt6 \
+		-DINSTALL_MKSPECSDIR=lib/qt6/mkspecs \
+		-DINSTALL_PUBLICBINDIR=${TERMUX_PREFIX}/opt/qt6/cross/bin \
+		-DQT_ALLOW_SYMLINK_IN_PATHS=OFF \
+		-DQT_FEATURE_freetype=ON \
+		-DQT_FEATURE_gui=ON \
+		-DQT_FEATURE_harfbuzz=ON \
+		-DQT_FEATURE_widgets=ON \
+		-DQT_FEATURE_zstd=OFF
+	ninja \
+		-j ${TERMUX_MAKE_PROCESSES} \
```

Comment:
ninja runs multiple jobs by-default. Isn't the `-j` option here redundant?

Reply:
`TERMUX_MAKE_PROCESSES` gives the dev a way to override the defaults and manually set how many cores they want used, which is useful when doing other things too.

Reply:
ninja's default n+2 jobs option sometimes is slower than n jobs option where n is total number of CPUs on host (based on my past experience). It is better to have an option to override it if the builder choose to do so.